Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -583,70 +584,71 @@ static bool sampleBufferRenderersSupportKeySession()
     ASSERT(!m_hasSessionSemaphore);
     ASSERT(!m_abortSemaphore);
 
-    m_parser->setDidParseInitializationDataCallback([weakThis = ThreadSafeWeakPtr { *this }] (InitializationSegment&& segment) {
-        ASSERT(isMainThread());
-        if (RefPtr protectedThis = weakThis.get())
-            protectedThis->didReceiveInitializationSegment(WTFMove(segment));
-    });
-
-    m_parser->setDidProvideMediaDataCallback([weakThis = ThreadSafeWeakPtr { *this }] (Ref<MediaSampleAVFObjC>&& sample, uint64_t trackId, const String& mediaType) {
```

Comment:
Changing where to set the callback does not seem to make a difference with regards to the crash.
It doesn't seem to harm either but calls the question of why doing so. Please add a comment in the change log.

Reply:
I did mention it in the commit log on the why : "We ensure that m_parser is only ever used on the parser queue."